pydantic/schema.py

```diff
@@ -587,7 +589,9 @@ def field_singleton_sub_fields_schema(
     (EmailStr, {'type': 'string', 'format': 'email'}),
     (UrlStr, {'type': 'string', 'format': 'uri'}),
     (DSN, {'type': 'string', 'format': 'dsn'}),
+    (SecretStr, {'type': 'string', 'writeOnly': True, 'format': 'secret-str'}),
```

Comment:
Great! I would suggest removing the `'format': 'secret-str'` and leaving only the `'writeOnly': True`.

This is because the `writeOnly` already has the meaning that it should be treated as a secret/password.

`format` is used to give additional semantic information about how to represent a value, and there are several pre-existing formats for that. And although `format` can be extended with custom values, in this case it wouldn't give any additional meaning.